src/Arcus.Messaging.Pumps.ServiceBus.KeyRotation/ReAuthenticateMessageHandler.cs

```diff
@@ -0,0 +1,72 @@
+        /// </summary>
+        /// <param name="messagePump">The message pump instance to restart when the message handler process an <see cref="SecretNewVersionCreated"/> event.</param>
+        /// <param name="logger">The logger instance to write diagnostic trace messages during the processing of the event.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="messagePump"/> or <paramref name="logger"/> is <c>null</c>.</exception>
+        public ReAuthenticateMessageHandler(AzureServiceBusMessagePump messagePump, ILogger<ReAuthenticateMessageHandler> logger)
+        {
+            Guard.NotNull(messagePump, nameof(messagePump), $"Requires an message pump instance to restart when the message handler process an {nameof(SecretNewVersionCreated)} event");
+            Guard.NotNull(logger, nameof(logger), "Requires an logger instance to write diagnostic trace messages during the processing of the event");
+
+            _messagePump = messagePump;
+            _logger = logger;
+        }
+
+        /// <summary>
+        ///     Process a new message that was received
+        /// </summary>
+        /// <param name="message">Message that was received</param>
+        /// <param name="messageContext">Context providing more information concerning the processing</param>
+        /// <param name="correlationInfo">
+        ///     Information concerning correlation of telemetry and processes by using a variety of unique
+        ///     identifiers
+        /// </param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="message"/> is <c>null</c>.</exception>
+        /// <exception cref="CloudException">Thrown when the <paramref name="message"/> doesn't represent an <see cref="SecretNewVersionCreated"/> event.</exception>
+        public async Task ProcessMessageAsync(
+            CloudEvent message,
+            AzureServiceBusMessageContext messageContext,
+            MessageCorrelationInfo correlationInfo,
+            CancellationToken cancellationToken)
+        {
+            Guard.NotNull(message, nameof(message), "Cannot invalidate Azure KeyVault secret from a 'null' CloudEvent");
+
+            _logger.LogTrace("Receiving new Azure Key Vault notification...");
+            
+            var secretNewVersionCreated = message.GetPayload<SecretNewVersionCreated>();
+            if (secretNewVersionCreated is null)
+            {
+                throw new CloudException(
+                    "Azure Key Vault job cannot map Event Grid event to CloudEvent because the event data isn't recognized as a 'SecretNewVersionCreated' schema");
+            }
+
+            _logger.LogInformation("Received Azure Key vault 'Secret New Version Created' event");
+            
+            await _messagePump.RestartAsync();
```

Comment:
Hold your horses 😁 You only need to restart it when the secret name matches the one that is being used to authenticate the message pump with 😅